### PR TITLE
Serviceaccounts: Add ability to add samename SA for different orgs

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
We are checking for login name across orgs when we create Service accounts, this implementation adds the orgID as a "middle prefix". 

**NOTE:** We could add it as a suffix instead. Just implementation details

https://github.com/grafana/grafana/issues/83889

migrations tested on:
- sqlite ✅ 
- mysql ✅ 
- postgres ✅ 

testing done via:
```bash
make devenv sources=postgres
make devenv sources=mysql
```

removed the migration, inserted a service account and rerun grafana.

### CODE REVIEW

File: pkg/services/user/userimpl/user.go [hunk 1]
```diff
@@ -406,6 +406,7 @@ func (s *Service) CreateServiceAccount(ctx context.Context, cmd *user.CreateUser
 	cmd.Email = cmd.Login
 	err := s.store.LoginConflict(ctx, cmd.Login, cmd.Email, s.cfg.CaseInsensitiveLogin)
 	if err != nil {
+		fmt.Printf("error %e", err)
```
Comment (eleijonmarck) [inline pkg/services/user/userimpl/user.go:409]:
```suggestion
```

File: pkg/services/serviceaccounts/database/store.go [hunk 2]
```diff
@@ -42,10 +42,19 @@ func ProvideServiceAccountsStore(cfg *setting.Cfg, store db.DB, apiKeyService ap
 	}
 }
 
+// generatedLogin makes a generated string to have a ID for the service account across orgs and it's name
+// this causes you to create a service account with the same name in different orgs
+// not the same name in the same org
+func generatedLogin(prefix string, orgId int64, name string) string {
```
Comment (Jguer) [inline pkg/services/serviceaccounts/database/store.go:48]:
```suggestion
func generateLogin(prefix string, orgId int64, name string) string {
```

**Review by Jguer — APPROVED** [review]
Very elegant solution :+1:

**Review by gamab — APPROVED** [review]

**Review by kalleep — COMMENTED** [review]
I wonder if we need migration for existing service account to follow this naming patter.

Otherwise you could use the same name for service accounts that was created before this right?

**Comment by eleijonmarck** [comment]
@kalleep yea, i guess we would have to migrate in this case. 

It could also be that we would "according" to terraform, create new service accounts. Since we no longer are able to retrieve the previous named service accounts.

We could make a migration where we do not migrate but duplicate the serviceaccounts. ugly solution.

Either way we change this it will impact customers

**Pull request merged**